src/main/java/songbox/house/util/JsonUtils.java

```diff
@@ -81,6 +84,30 @@ public static VkSearchResponseDto searchFromNewsFeedResponseToObject(final Strin
     }
 
     public static <T> T responseToObject(final String body, final Class<T> clazz) {
+        // Try #1
+        try {
```

Comment:
Maybe create 2 methods:
1) private static <T> Optional<T> parse(final String body, final Class<T> clazz)
2) private static <T> T parseWithOldApi(final String body, final Class<T> clazz)
And change this public method body to
`return parse(body, clazz).orElseGet(() -> parseWithOldApi(body, clazz));`

With this implementation your comments "Try #1" and "Try Old api" will be redundant. Ideally, the code should be self-documented :)
Anyway, it's just syntax sugar and your solution is cool)

Reply:
Done 😉